eng/scripts/New-ChangelogEntry.ps1

```diff
@@ -0,0 +1,314 @@
+}
+
+# Get repository root
+$repoRoot = Split-Path (Split-Path $PSScriptRoot -Parent) -Parent
+$changelogEntriesDir = Join-Path $repoRoot $ChangelogEntriesPath
+$schemaPath = Join-Path $repoRoot "eng/schemas/changelog-entry.schema.json"
+
+# Create changelog-entries directory if it doesn't exist
+if (-not (Test-Path $changelogEntriesDir)) {
+    Write-Host "Creating changelog-entries directory: $changelogEntriesDir" -ForegroundColor Yellow
+    New-Item -ItemType Directory -Path $changelogEntriesDir | Out-Null
+}
+
+# Determine if we're in interactive mode (missing Description, Section, or PR)
+$isInteractive = (-not $Description) -or (-not $Section) -or (-not $PSBoundParameters.ContainsKey('PR'))
+
+# Interactive mode if parameters not provided
+if (-not $Description) {
+    Write-Host "`nChangelog Entry Creator" -ForegroundColor Cyan
+    Write-Host "======================" -ForegroundColor Cyan
+    Write-Host ""
+    Write-Host "Note: For multi-line descriptions (e.g., with lists), use the -Description parameter with a here-string." -ForegroundColor Gray
+    Write-Host ""
+    
+    $Description = Read-Host "Description (minimum 10 characters)"
+    # Trim whitespace from user input
+    $Description = $Description.Trim()
+    
+    while ($Description.Length -lt 10) {
+        Write-Host "Description must be at least 10 characters long." -ForegroundColor Red
+        $Description = Read-Host "Description (minimum 10 characters)"
+        $Description = $Description.Trim()
+    }
+}
+
+if (-not $Section) {
+    Write-Host "`nSelect a section:" -ForegroundColor Cyan
+    Write-Host "1. Features Added"
+    Write-Host "2. Breaking Changes"
+    Write-Host "3. Bugs Fixed"
+    Write-Host "4. Other Changes"
+    
+    $choice = Read-Host "`nEnter choice (1-4)"
+    $Section = switch ($choice) {
+        "1" { "Features Added" }
+        "2" { "Breaking Changes" }
+        "3" { "Bugs Fixed" }
+        "4" { "Other Changes" }
+        default {
+            Write-Host ""
+            Write-Host "ERROR: Invalid choice '$choice'. Please select 1-4." -ForegroundColor Red
+            Write-Host ""
+            exit 1
+        }
+    }
+}
+
+# Allow subsection for any section in interactive mode
+if (-not $PSBoundParameters.ContainsKey('Subsection') -and $isInteractive) {
+    $subsectionInput = Read-Host "`nSubsection (optional, press Enter to skip)"
+    if ($subsectionInput) {
+        # Trim and title case the subsection
+        $Subsection = ConvertTo-TitleCase -Text $subsectionInput.Trim()
+    }
+}
+
+if (-not $PR -and $isInteractive) {
+    $prInput = Read-Host "`nPR number (press Enter to skip if not known yet)"
+    if ($prInput) {
+        $PR = [int]$prInput
+    }
+}
+
+# Trim whitespace from description if provided via parameter
+$Description = $Description.Trim()
+
+# Generate filename with timestamp in milliseconds
+$timestamp = [DateTimeOffset]::UtcNow.ToUnixTimeMilliseconds()
+$filename = "$timestamp.yml"
```

Comment:
I think this will make it hard to find any specific unmerged entry.  I like the expressive names used here:
https://github.com/tauri-apps/tauri/tree/dev/.changes
```
add-plugin-listener-error-handling.md
fix-needles-collect.md
nsis-3.11.md
perf-remove-needless-clone.md
pnpm-package-version-check.md
version-req-error.md
```